src/silx/math/histogramnd/src/histogramnd_template.c

```diff
@@ -36,7 +36,7 @@ int TEMPLATE(histogramnd, HISTO_SAMPLE_T, HISTO_WEIGHT_T, HISTO_CUMUL_T)
                         (HISTO_SAMPLE_T *i_sample,
                          HISTO_WEIGHT_T *i_weights,
                          int i_n_dim,
-                         int i_n_elem,
+                         size_t i_n_elem,
```

Comment:
For consistency you'd better also change the type of `elem_idx` below:
```c
long elem_idx = 0;
```

Reply:
Thanks, I updated it.  
In `chistogramnd_lut.pyx` however, `elem_idx` should be left as `long` as it can be negative:

```C
elem_idx = 0 - i_n_dims
```